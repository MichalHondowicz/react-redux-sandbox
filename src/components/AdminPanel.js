import React from 'react';
import LoginForm from './LoginForm';
import BookForm from './BookForm';
import AdminInventoryList from "./AdminInventoryList";
import {base} from "../base";
import {Link} from "react-router-dom";


class AdminPanel extends React.Component {

    constructor() {
        super();
        this.state = {
            loggedIn: false
        }
    }

    handleLoggedState = (logger) => {
        this.setState({loggedIn: logger})
    };

    addNewBook = (newBook) => {
        if (Array.isArray(this.state.books)) {
            this.setState({
                books: [...this.state.books, newBook],
            });
        } else {
            this.setState({books: [newBook]})
        }
    };

    editBook = (previousBook, updatedBook) => {

        const newBooks = this.state.books.filter(book => previousBook !== book.name);

        this.setState({
            books: [...newBooks, updatedBook],
        });
    };

    removeFromInventory = (inventoryItem) => {
        this.setState({
            books: this.state.books.filter(book => inventoryItem !== book)
        })
    };

    componentDidMount() {
        this.ref = base.syncState('bookstore/books', {
            context: this,
            state: 'books'
        })
    }

    componentWillUnmount() {
        base.removeBinding(this.ref)
    }

    render() {
        return (
            <div>
                {!this.state.loggedIn &&
                <LoginForm handleLoggedState={this.handleLoggedState}/>
                }
                {this.state.loggedIn &&
                <React.Fragment>
                    <BookForm handleLoggedState={this.handleLoggedState} addNewBook={this.addNewBook}
                              editBook={this.editBook}/>
                    <AdminInventoryList books={this.state.books} removeFromInventory={this.removeFromInventory}
                    />
                </React.Fragment>
                }
                <Link to='/'>
                    <button className='ui secondary button'>Home</button>
                </Link>
            </div>
        )
    }
}

export default AdminPanel;