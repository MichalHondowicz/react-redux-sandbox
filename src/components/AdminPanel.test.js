import ReactDOM from "react-dom";
import AdminPanel from "./AdminPanel";
import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

configure({adapter: new Adapter()});

describe('AdminPanel test', () => {
    it('AdminPanel renders with no problems ', () => {
        const div = document.createElement('div');
        ReactDOM.render(<AdminPanel/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Snapshot ok', () => {
        const wrapper = shallow(<AdminPanel/>);
        expect(wrapper).toMatchSnapshot();
    })
});