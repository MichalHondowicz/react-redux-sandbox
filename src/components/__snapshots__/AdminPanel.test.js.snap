// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminPanel test Snapshot ok 1`] = `ShallowWrapper {}`;
